### PR TITLE
BFF 646 Build download yearly overview feature

### PR DESCRIPTION
**Pull request by Daniela510**
## Description

Added a download button to the AppBar on donations-history.
It builds a bottom sheet where you can pick a year, and request your yearly overview.
This is an api call. The actual overview is sent to your email. 

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: lib/features/overview/bloc/givt_bloc.dart [hunk 1]
```diff
@@ -225,4 +227,28 @@ class GivtBloc extends Bloc<GivtEvent, GivtState> {
...
+  ) async {
+    emit(const GivtLoading());
+    try {
+      await givtRepository.downloadYearlyOverview(
+        event.guid,
+        event.year,
+      );
+
+      emit(const GivtDownloadedSuccess());
+      add(const GivtInit());
```
Comment (alexbejann) [inline lib/features/overview/bloc/givt_bloc.dart:243]: Why is this needed?

File: lib/features/overview/pages/overview_page.dart [hunk 2]
```diff
@@ -290,6 +335,104 @@ class OverviewPage extends StatelessWidget {
...
+            ),
+          )
+          .toList(),
+      onChanged: (String? newValue) {
+        setState(() {
+          _overviewYearController = newValue!;
+        });
+      },
+    );
+  }
```
Comment (alexbejann) [inline lib/features/overview/pages/overview_page.dart:434]: Can't you extract this into a separate widget and use a callback to get the selected year? In this way you don't need to convert the whole widget into a stateful one

File: lib/shared/repositories/givt_repository.dart [hunk 3]
```diff
@@ -107,4 +109,9 @@ class GivtRepositoryImpl with GivtRepository {
   @override
   Future<bool> deleteGivt(List<dynamic> ids) async =>
       apiClient.deleteGivts(body: ids);
+
+  @override
+  Future<void> downloadYearlyOverview(String guid, String year) async {
+    await apiClient.downloadYearlyOverview(guid, year);
```
Comment (alexbejann) [inline lib/shared/repositories/givt_repository.dart:115]: isn't this supposed to return a bool as in the "apiClient.downloadYearlyOverview(guid, year)"?

**Review by alexbejann — CHANGES_REQUESTED** [review]

File: lib/features/overview/bloc/givt_bloc.dart (hunk 1 above)
Comment (Daniela510) [reply lib/features/overview/bloc/givt_bloc.dart]: Emiting GivtLoading & GivtDownload success erased the givts from state, so i needed to refetch. 
I changed it so GivtDownloadedSuccess carries the givts from prev state.

File: lib/features/overview/pages/overview_page.dart (hunk 2 above)
Comment (Daniela510) [reply lib/features/overview/pages/overview_page.dart]: Done

File: lib/shared/repositories/givt_repository.dart (hunk 3 above)
Comment (Daniela510) [reply lib/shared/repositories/givt_repository.dart]: Yeah

File: lib/features/overview/pages/overview_page.dart
Comment (alexbejann) [inline lib/features/overview/pages/overview_page.dart:1]: @Daniela510 could you change the widget back to a stateless widget?

File: lib/features/overview/widgets/dropdown_year_menu.dart
Comment (alexbejann) [inline lib/features/overview/widgets/dropdown_year_menu.dart:1]: Nicee!

**Review by alexbejann — APPROVED** [review]
@Daniela510 Just change the [overview page](lib/features/overview/pages/overview_page.dart) to a stateless widget
